main.cpp

```diff
@@ -66,18 +66,22 @@ struct MbedClientDevice device = {
 MbedClient mbed_client(device);
 
 
-// In case of K64F board , there is button resource available
-// to change resource value and unregister
-#ifdef TARGET_K64F
+// If board has a button_1, use it to update the counter.
+#ifdef BUTTON1
```

Comment:
!!#!"¤#! ENUMs. They are almost useless, because they can't be used via #ifdefs.
We need another solution then, or a PR to mbed OS to change this. Not sure if that's very feasible...